### ISSUE TITLE
Handle closed selector exception

### ISSUE THREAD

**Issue by gregw** [eclipse/jetty.project#3580]
An instance has been reported that continually logs the exception below.   While it is unclear how/why this occurred, we should inspect the code to see if we can recover from such exceptions or perhaps make them more fatal.

One issue is that if only a single selector suffers from this problem, health checks on the server may still work on another selector and thus keep the instance alive.
 
```
Caused by: java.nio.channels.ClosedSelectorException
        at sun.nio.ch.SelectorImpl.keys(SelectorImpl.java:68)
        at org.eclipse.jetty.io.ManagedSelector.size(ManagedSelector.java:104)
        at org.eclipse.jetty.io.SelectorManager.chooseSelector(SelectorManager.java:190)
        at org.eclipse.jetty.io.SelectorManager.connect(SelectorManager.java:207)
        at org.eclipse.jetty.client.AbstractHttpClientTransport.connect(AbstractHttpClientTransport.java:118)
```

**Comment by sbordet** [comment]
See also #4798.

**Comment by gregw** [comment]
closed as duplicated by #4798 with more discussion there.
